api/krusty/remoteloader_test.go

```diff
@@ -0,0 +1,444 @@
+  containers:
+  - image: nginx:1.7.9
+    name: nginx
+`,
+		},
+		{
+			name: "has ref path timeout and origin annotation",
+			kustomization: `
+resources:
+- file://$ROOT/multibase.git/dev?version=main&timeout=500
+buildMetadata: [originAnnotations]
+`,
+			expected: `apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    config.kubernetes.io/origin: |
+      path: base/pod.yaml
+      repo: file://$ROOT/multibase.git
+      ref: main
+  labels:
+    app: myapp
+  name: dev-myapp-pod
+spec:
+  containers:
+  - image: nginx:1.7.9
+    name: nginx
+`,
+		},
+		{
+			name: "repo does not exist ",
+			kustomization: `
+resources:
+- file:///not/a/real/repo
+`,
+			err: "fatal: '/not/a/real/repo' does not appear to be a git repository",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if test.skip {
+				t.SkipNow()
+			}
+
+			kust := strings.ReplaceAll(test.kustomization, "$ROOT", repos.root)
+			fSys, tmpDir := createKustDir(t, kust)
+
+			b := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
+			m, err := b.Run(
+				fSys,
+				tmpDir.String())
+
+			if test.err != "" {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), test.err)
+			} else {
+				require.NoError(t, err)
+				checkYaml(t, m, strings.ReplaceAll(test.expected, "$ROOT", repos.root))
+			}
+		})
+	}
+}
+
+func TestRemoteLoad_RemoteProtocols(t *testing.T) {
+	// Slow remote tests with long timeouts.
+	// TODO: If these end up flaking, they should retry. If not, remove this TODO.
+	tests := []struct {
+		name          string
+		kustomization string
+		err           string
+		errT          error
+		beforeTest    func(t *testing.T, )
+	}{
+		{
+			name: "https",
+			kustomization: `
+resources:
+- https://github.com/kubernetes-sigs/kustomize//examples/multibases/dev/?submodules=0&ref=kustomize%2Fv4.5.7&timeout=300
+`,
+		},
+		{
+			name: "git double-colon https",
+			kustomization: `
+resources:
+- git::https://github.com/kubernetes-sigs/kustomize//examples/multibases/dev/?submodules=0&ref=kustomize%2Fv4.5.7&timeout=300
+`,
+		},
+		{
+			name: "https raw remote file",
+			kustomization: `
+resources:
+- https://raw.githubusercontent.com/kubernetes-sigs/kustomize/v3.1.0/examples/multibases/base/pod.yaml?timeout=300
+namePrefix: dev-
+`,
+		},
+		{
+			name:       "ssh",
+			beforeTest: configureGitSSHCommand,
+			kustomization: `
+resources:
+- git@github.com/kubernetes-sigs/kustomize/examples/multibases/dev?submodules=0&ref=kustomize%2Fv4.5.7&timeout=300
+`,
+		},
+		{
+			name:       "ssh with colon",
+			beforeTest: configureGitSSHCommand,
+			kustomization: `
+resources:
+- git@github.com:kubernetes-sigs/kustomize/examples/multibases/dev?submodules=0&ref=kustomize%2Fv4.5.7&timeout=300
+`,
+		},
+		{
+			name:       "ssh without username",
+			beforeTest: configureGitSSHCommand,
+			kustomization: `
+resources:
+- github.com/kubernetes-sigs/kustomize/examples/multibases/dev?submodules=0&ref=kustomize%2Fv4.5.7&timeout=300
```

Comment:
Are you sure this is interpreted as SSH? I thought we default the protocol to `https`, and indeed if I run this as input in a repo spec test, https is added to the clone spec.

Reply:
Hmm I thought I saw this one was using SSH and changed the test that way. I'll change it back.